File: CPP/CPP Adv/Exception Handle/DefinedException.cpp
// cpp - UserDefined Exception
#include <iostream>
class CustomException : public std::exception
{
private:
    /* data */
public:
    CustomException(/* args */);
    ~CustomException();
};

CustomException::CustomException(/* args */)
{
    printf("CustomException : Divide by zero Error ! \n");
}

CustomException::~CustomException()
{
}

int main(int argc, char const *argv[])
{
    int a = 1, b = 0;
    try
    {
        if (b == 0)
            throw CustomException();
        else
            printf("%d", a / b);
    }
    catch (const CustomException &e)
    {
        printf("Catch block : Error Occured !\n");
    }

    return 0;
}

File: Java/Java Adv/Exception Handeling/DefinedException.java
public class DefinedException {
    public float inverse(int num) throws Custom_Exception {
        return (1 / num);
    }

    public static void main(String[] args) {
        try {
            System.out.println(divide(0));
        } catch (Exception e) {
            // TODO: handle exception
            System.out.println("Error Occured !");
        }
    }
}

public class Custom_Exception extends Exception {
    public Custom_Exception(String Message) {
        super(Message);
    }
}